sdk/storage/azblob/zc_storage_error.go

```diff
@@ -70,22 +73,32 @@ func handleError(err error) error {
 }
 
 // defunkifyStorageError is a function that takes the "funky" ResponseError and reduces it to a storageError.
-func defunkifyStorageError(responseError ResponseError) error {
-	if err, ok := responseError.Unwrap().(*StorageError); ok {
-		// errors.Unwrap(responseError.Unwrap())
+func defunkifyStorageError(responseError *azcore.ResponseError) error {
+	var storageError StorageError
+	body, err := runtime.Payload(responseError.RawResponse)
+	if err != nil {
+		goto Default
+	}
+	if len(body) > 0 {
+		if err := xml.Unmarshal(body, &storageError); err != nil {
+			goto Default
+		}
+	}
 
-		err.response = responseError.RawResponse()
+	// errors.Unwrap(responseError.Unwrap())
```

Comment:
If unwrapping is no longer necessary, then this line should be removed (not commented out).

Reply:
This was a preexisting artifact.  I can remove it, but I prefer to leave general code hygiene to the owning team.

Reply:
Fixed